sovrin_client/anon_creds/sovrin_public_repo.py

```diff
@@ -54,30 +54,54 @@ def __init__(self, client, wallet):
         self.displayer = print
 
     async def getSchema(self, id: ID) -> Optional[Schema]:
-        op = {
-            TARGET_NYM: id.schemaKey.issuerId,
-            TXN_TYPE: GET_SCHEMA,
-            DATA: {
-                NAME: id.schemaKey.name,
-                VERSION: id.schemaKey.version,
+        data = None
+        if id.schemaKey:
+            op = {
+                TARGET_NYM: id.schemaKey.issuerId,
+                TXN_TYPE: GET_SCHEMA,
+                DATA: {
+                    NAME: id.schemaKey.name,
+                    VERSION: id.schemaKey.version,
+                }
             }
-        }
-        data, seqNo = await self._sendGetReq(op)
+            data, seqNo = await self._sendGetReq(op)
+
+        else:
+            op = {
+                TXN_TYPE: GET_TXNS,
+                DATA: id.schemaId
+            }
+            res, seqNo = await self._sendGetReq(op)
+            if res:
+                data = json.loads(res[DATA]) if res else {}
+                data[ORIGIN] = res[IDENTIFIER]
+
         return Schema(name=data[NAME],
                       version=data[VERSION],
                       attrNames=data[ATTR_NAMES],
                       issuerId=data[ORIGIN],
                       seqId=seqNo) if data else None
 
-    async def getPublicKey(self, id: ID,
-                           signatureType = 'CL') -> Optional[PublicKey]:
-        op = {
-            TXN_TYPE: GET_CLAIM_DEF,
-            REF: id.schemaId,
-            ORIGIN: id.schemaKey.issuerId,
-            SIGNATURE_TYPE: signatureType
-        }
-        data, seqNo = await self._sendGetReq(op)
+    async def getPublicKey(self, id: ID = None,
```

Comment:
Please either remove `seqId` or split it into two method (`getPublicKey` and `getPublicKeyBySeqId`)